.github/workflows/book.yml

```
@@ -0,0 +1,38 @@
+name: deploy-book
+
+# Only run this when the master branch changes
+on:
+  push:
+    paths:
+      - 'docs/**'
+    branches:
+      - 'master'
+
+# This job installs dependencies, build the book, and pushes it to `gh-pages`
+jobs:
+  deploy-book:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Install dependencies
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+
+      - name: Install dependencies
+        run: |
+          pip install -r docs/requirements.txt
+
+      # Build the book
+      - name: Build the book
+        run: |
+          jupyter-book build docs/
+
+      # Push the book's HTML to github-pages
+      - name: GitHub Pages action
+        uses: peaceiris/actions-gh-pages@v3.5.9
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: docs/_build/html
```

A suggestion: maybe publish to a [separate branch](https://github.com/peaceiris/actions-gh-pages/blob/main/README.md#%EF%B8%8F-set-another-github-pages-branch-publish_branch) instead?

```suggestion
          publish_branch: gh-pages
```

I beileve this would help keep the master branch (or any "code" branch) cleaner.